sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala

```diff
@@ -95,6 +106,38 @@ object ColumnarPartitionReaderWithPartitionValues {
     result
   }
 
+  def addGpuColumVectorsToBatchIter(
+     fileBatch: ColumnarBatch,
+     partitionColumnVectors: Array[Array[GpuColumnVector]]): Iterator[ColumnarBatch] = {
+    val fileBatchCols = (0 until fileBatch.numCols).map(fileBatch.column)
+
+    // Calculate the cumulative row counts for splitting the GPU ColumnVectors.
+    // Note:
+    //  1. ai.rapids.cudf.ColumnView#split only accepts 'int' indices
+    //  2. Split indices must be in cumulative format
+    val cumulativeRowCounts = partitionColumnVectors.scanLeft(0) {
+      case (cumulativeCount, partitionColumns) =>
+        cumulativeCount + partitionColumns.head.getRowCount.toInt
+    }.drop(1).dropRight(1)
+    // Split each column into multiple batches based on the split indices
+    val splitColumnVectors = fileBatchCols.map {
+      case gpuColumnVector: GpuColumnVector =>
+        val splitColumns = gpuColumnVector.getBase.split(cumulativeRowCounts: _*)
+        splitColumns.map(splitCol => GpuColumnVector.from(splitCol, gpuColumnVector.dataType()))
+    }
+
+    // Combine the split GPU ColumnVectors with partition ColumnVectors.
+    partitionColumnVectors.zipWithIndex.map {
+      case (partitionColumns, index) =>
+        val splitCols = splitColumnVectors.map(col => col(index))
+        val resultCols = splitCols ++ partitionColumns
+        val batchNumRows = resultCols.headOption.map(_.getRowCount.toInt).getOrElse(0)
```

Comment:
similar here, we recalculate row count again but don't we already know this number? seem less expensive to just reuse that then recalculate.